### PR TITLE
$path

### PR DESCRIPTION
**Pull request by gausie**
This does most of the move to $path, but we need to refactor `Paths.ts`. I think the easiest thing to do is make utils called canUseSkillBooks(path), hasALlPerms(path) etc etc

### CODE REVIEW

**Review by lewismd13 — APPROVED** [review]

File: src/ascend.ts [hunk 1]
```diff
@@ -222,7 +222,7 @@ export function ascend(
   pet: Item | undefined = undefined,
   permSkills: Map<Skill, Lifestyle> | undefined = undefined
 ): void {
-  if (!path.classes.includes(playerClass)) {
+  if (playerClass.path !== path) {
```
Comment (horrible-little-slime) [inline src/ascend.ts:225]:
```suggestion
  if (path === Path.none || (path.avatar && playerClass.path !== path))
```

Comment (horrible-little-slime) [reply src/ascend.ts:225]: I am 80% sure this is what we want

Comment (gausie) [reply src/ascend.ts:225]: oy vey

Comment (gausie) [reply src/ascend.ts:225]: So maybe you want to do
```ts
if (path.avatar && playerClass.path !== path)
```
?

Comment (horrible-little-slime) [reply src/ascend.ts:225]: That doesn't work. What if I try to ascend journeyman as ed?

I also think we should make sure you can't ascend as Path.none

**Pull request merged**